### PR TITLE
Move jboss find commands to separate playbook and make non-default

### LINKED ISSUE #336: Remove find commands from default JBoss EAP scan role
Part of #335 .

### PR DESCRIPTION
**Pull request by noahl**
Two separate steps here:
  - Moving the JBoss find commands to their own playbook
  - Removing the scan facts from the default fact list

This has the result that Rho will not run a scan unless specifically
requested, and also the find commands are visibly separated from the
rest.

Closes #336 .

### CODE REVIEW

File: rho_playbook.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@
     - virt
     - virt_what
     - jboss
+    - jboss-scan
```
Comment (chambridge) [inline rho_playbook.yml]: I've never liked the role name of `jboss` when its more focused on *EAP*. If we are naming/renaming things could we update it to have an eap focus?

File: roles/jboss-scan/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+---
+# Scan linux systems for JBoss EAP or Wildfly Installations
+    - name: Gather jboss.jar-ver
+      raw: FOUND=""; for jar in `find {{scan_dirs}} -xdev -name 'jboss-modules.jar' 2>/dev/null | grep -v '\.installation/patches'`; do VERSION=$(java -jar ${jar} -version 2> /dev/null | grep version | sed 's/.*version\s//g'); inode=$(stat -c '%i' "${jar}"); fs=$(df  -T "${jar}" | grep "/dev" | sed 's/ .*//'); ctime=$(stat ${jar} | grep 'Change' | grep -oP '[1-2][0-9]{3}-[0-1][0-9]-[0-3][0-9]'); if [ ! -z "${VERSION}" ]; then if [ ! -z "$FOUND" ]; then FOUND="$FOUND; $VERSION**$ctime"; else FOUND=${VERSION}'**'${ctime}; fi; fi; done; echo ${FOUND}
+      register: jboss.jar-ver
```
Comment (chambridge) [inline roles/jboss-scan/tasks/main.yml]: wouldn't mind updating the fact names either to be EAP focus, but that could be for another PR.

**Review by chambridge — APPROVED** [review]
A few comments based on preference. But fine with this otherwise.

File: rho_playbook.yml (hunk 1 above)
Comment (noahl) [reply rho_playbook.yml]: Good point! I'll rename the fact to `jboss.eap.*` as we discussed. The folder can be something like `jboss-eap` or `jboss_eap`, unless you have a preference here.

File: roles/jboss-scan/tasks/main.yml (hunk 2 above)
Comment (noahl) [reply roles/jboss-scan/tasks/main.yml]: Will do.

File: rho/utilities.py [hunk 3]
```diff
@@ -113,10 +113,10 @@
                       'subman.virt.uuid',
                       'subman.has_facts_file')
 
-JBOSS_FACTS_TUPLE = ('jboss.running-versions',)
+JBOSS_FACTS_TUPLE = ('jboss.eap.running-versions',)
```
Comment (chambridge) [inline rho/utilities.py]: Updates needed in `doc/source/command_syntax_usage.rst` with changed fact names.
